Mathlib/CategoryTheory/Galois/Basic.lean

```diff
@@ -178,6 +180,14 @@ lemma not_initial_iff_fiber_nonempty (X : C) : (IsInitial X → False) ↔ Nonem
 lemma not_initial_of_inhabited {X : C} (x : F.obj X) (h : IsInitial X) : False :=
   ((initial_iff_fiber_empty F X).mp ⟨h⟩).false x
 
+/-- An object that is neither initial or connected has a non-trivial subobject. -/
+lemma has_non_trivial_subobject_of_not_isConnected_of_not_initial (X : C) (hc : ¬ IsConnected X)
+    (hi : IsInitial X → False) :
+    ∃ (Y : C) (v : Y ⟶ X), (IsInitial Y → False) ∧ Mono v ∧ (¬ IsIso v) := by
+  by_contra h2
+  simp only [exists_and_left, not_exists, not_and, not_not] at h2
+  exact hc ⟨hi, fun Y i hm hni ↦ h2 Y hni i hm⟩
```

Comment:
Does this work
```suggestion
  contrapose! hc
  exact ⟨hi, fun Y i hm hni ↦ hc Y hni i hm⟩
```

Reply:
Indeed, this works! I did not know the `!` option, that is quite neat.